javascript-typing-tutor/main.js

```diff
@@ -0,0 +1,23 @@
+var $span = document.querySelectorAll('span');
+
+document.addEventListener('keydown', function keydown(event) {
+  var $keypressed = event.key;
+  for (var i = 0; i < $span.length; i++) {
+    if ($keypressed === $span[i].textContent) {
+      if ($span[i].className === 'next') {
+        $span[i].className = 'correct';
+        $span[++i].className = 'next';
+      }
+    } else {
+      if ($span[i].className === 'next') {
+        $span[i].className = 'wrong';
+      } else {
+        if ($span[i].className === 'wrong') {
+          if ($keypressed === 'Backspace') {
+            $span[i].className = 'next';
+          }
+        }
+      }
+    }
+  }
```

Comment:
You shouldn't need to loop over every letter every time somebody types a character, you will only need to do deal with the element with the `next` class and possibly the element after it if the letter was correct and you need to make the next letter have the `next` class. I would suggest making a variable to keep track of the index of the current letter, and just increment that value by 1 if the user got a letter correct.